lisa/tools/python.py

```diff
@@ -89,3 +94,79 @@ def exists_package(self, package_name: str) -> bool:
     def uninstall_package(self, package_name: str) -> bool:
         result = self.run(f"uninstall {package_name} -y", force_run=True, sudo=True)
         return result.exit_code == 0
+
+
+class PythonVenv(Tool):
+    @property
+    def command(self) -> str:
+        return f"{self.get_venv_path()}/bin/{self.python.command}"
+
+    @property
+    def can_install(self) -> bool:
+        return True
+
+    @property
+    def dependencies(self) -> List[Type[Tool]]:
+        return [Python]
+
+    def _create_venv(self, venv_path: str = "", sudo: bool = False) -> PurePath:
+        _venv_path: PurePath = (
+            PurePath(venv_path) if venv_path else self.node.working_path / "venv"
+        )
+        cmd_result = self.python.run(
+            f"-m venv {_venv_path}", force_run=True, sudo=sudo, shell=True
+        )
+        assert_that(
+            cmd_result.exit_code, f"fail to create venv: {_venv_path}"
+        ).is_equal_to(0)
+        self._venv_path = _venv_path
+        return self._venv_path
+
+    def __init__(self, node: "Node", venv_path: str = "") -> None:
+        super().__init__(node)
+        self.python: Python = self.node.tools[Python]
```

Comment:
use `self._python`, because it's a private variable.

Reply:
updated